extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeMixPlaylistExtractorTest.java

```diff
@@ -0,0 +1,225 @@
+                        + index);
+            extractor.fetchPage();
+        }
+
+        @Test
+        public void getName() throws Exception {
+            String name = extractor.getName();
+            assertThat(name, startsWith("Mix"));
+            assertThat(name, containsString(videoTitle));
+        }
+
+        @Test
+        public void getThumbnailUrl() throws Exception {
+            final String thumbnailUrl = extractor.getThumbnailUrl();
+            assertIsSecureUrl(thumbnailUrl);
+            assertThat(thumbnailUrl, containsString("yt"));
+            assertThat(thumbnailUrl, containsString(videoId));
+        }
+
+        @Test
+        public void getNextPageUrl() throws Exception {
+            final String nextPageUrl = extractor.getNextPageUrl();
+            assertIsSecureUrl(nextPageUrl);
+            assertThat(nextPageUrl, containsString("list=RD" + videoId));
+        }
+
+        @Test
+        public void getInitialPage() throws Exception {
+            InfoItemsPage<StreamInfoItem> streams = extractor.getInitialPage();
+            assertThat(streams.getItems().isEmpty(), is(false));
+            assertThat(streams.hasNextPage(), is(true));
+        }
+
+        @Test
+        public void getPage() throws Exception {
+            InfoItemsPage<StreamInfoItem> streams = extractor.getPage(extractor.getNextPageUrl());
+            assertThat(streams.getItems().isEmpty(), is(false));
+            assertThat(streams.hasNextPage(), is(true));
+        }
+
+        @Test
+        public void getPageMultipleTimes() throws Exception {
+            InfoItemsPage<StreamInfoItem> streams = extractor.getPage(extractor.getNextPageUrl());
+
+            //Should work infinitely, but for testing purposes only 3 times
+            for (int i = 0; i < 3; i++) {
+                assertThat(streams.hasNextPage(), is(true));
+                streams = extractor.getPage(streams.getNextPageUrl());
+            }
+        }
+    }
+
+    public static class MyMix {
+
+        @Before
+        public void setUp() throws Exception {
+            NewPipe.init(DownloaderTestImpl.getInstance());
+            extractor = (YoutubeMixPlaylistExtractor) YouTube
+                .getPlaylistExtractor(
+                    "https://www.youtube.com/watch?v=" + videoId + "&list=RDMM" + videoId);
+            extractor.fetchPage();
+        }
+
+        @Test
+        public void getServiceId() {
+            assertEquals(YouTube.getServiceId(), extractor.getServiceId());
+        }
+
+        @Test
+        public void getName() throws Exception {
+            String name = extractor.getName();
+            assertEquals("My Mix", name);
+        }
+
+        @Test
+        public void getThumbnailUrl() throws Exception {
+            final String thumbnailUrl = extractor.getThumbnailUrl();
+            assertIsSecureUrl(thumbnailUrl);
+            assertThat(thumbnailUrl, startsWith("https://i.ytimg.com/vi/_AzeUSL9lZc"));
```

Comment:
Because the extractor is set up with a starting video (watch?v=videoId). The rest of the mix is indeed different for everyone, but because there is a starting video, the first one will always be that one.

Reply:
Oh, that's right, sorry ;-)